sip/sip-001-burn-election.md

```diff
@@ -0,0 +1,1074 @@
+
+While there is no way to prevent this "burn collapse"
+scenario outright, the Stacks protocol helps reduce the likelihood of it
+happening through two mechanisms.  First, Stacks does not implement an all-or-nothing reward system --
+peers can hedge their bets by committing to 
+a distribution of chain tips, so they only need to send one burn transaction per chain tip candidate
+in order to receive _some_ reward instead of
+feeling compelled to one-up each other.  Second, the fork selection rules
+(described below) ensure that burns can only help leader so much -- they stop helping after a certain
+threshold, and too many burns over a short time hurts the leader's chances in future
+elections.
+
+# Reward distribution
+
+New Stacks tokens come into existence on a fork in an epoch where a leader is
+selected, and are granted to the leader if the leader produces a valid block.
+However, the Stacks blockchain pools all tokens created and all transaction fees received and
+does not distribute them until a large number of epochs (a _lockup period_) has
+passed.  The tokens cannot be spent until the period passes.
+
+The idealized payout distribution in a proof-of-burn blockchain is a
+distribution where each burner receives tokens proportional to the fraction of burns they
+contributed for the block.  In addition to implementing a lockup period, the
+Stacks protocol "smooths out" the token rewards over a _reward window_
+to help realize a close approximation of this payout distribution.
+
+## Sharing the rewards among winners
+
+Winning blocks and the leaders and users who burned for them are not rewarded
+in a winner-take-all fashion.  Instead, after the
+rewards are delayed for a lock-up period, the are distributed to all winning
+burns over a _reward window_.  The block rewards
+are allotted to each burning participant incrementally as the window passes,
+based on the ratio between how
+much it burned over the window versus how much everyone
+burned over the window.  This has the (desired) effect of "smoothing out" the
+rewards so as to minimize the difference between the idealized earnings and actual earnings
+across all winning participants.
+
+The "smoothing" mechanism is necessary to keep peers incentivized to mine a
+maximal number of transactions when either the Stacks block rewards
+or the burn chain block rewards are dominated by transaction fees instead
+of coinbases.  On the Stacks chain, the fact that the peers all share the transaction
+fees proportionally means that they are incentivized to mine a maximal number of
+transactions during their epochs, no matter how big they are relative to the
+coinbase.  This avoids some of the [chain instability](http://randomwalker.info/publications/mining_CCS.pdf)
+problems that arise in winner-takes-all payout schemes, whereby
+miners implement a "petty compliant" strategy where a minimal amount of
+transactions are mined per block once transaction fees exceed coinbase rewards.
```

Comment:
It isn't clear to me that this isn't still a "winner-take-all" event for a particular Epoch --- there's still only 1 winner for an epoch, its just that epoch rewards are being smoothed so that one epoch is never much more valuable to win than another epoch close to it.

The burn-to-mint behavior described in the token paper doesn't have this property. In that, the coinbase is distributed to all _valid_ burns in a window (not just winners) --  I just want to be sure what kind of behavior we're trying to
implement here.

Reply:
You're right, it is still winner-take-all in the sense that we don't reward leaders who don't win sortition.

However, I expect that leaders will also commit some "user support burns" behind other competing chain tips so they get some payout even if they don't win.  Wouldn't this address the problem?

Reply:
Okay -- after thinking this through, this kind of "uncle mining" has other negative consequences, and I think the benefits are sufficiently realized via the user miner support.